src/server/engine_shard_set.cc

```diff
@@ -545,29 +540,36 @@ void EngineShard::PollExecution(const char* context, Transaction* trans) {
     DVLOG(1) << "Skipped TxQueue " << continuation_trans_;
   }
 
-  // we need to run trans if it's OOO or when trans is blocked in this shard.
-  bool should_run = trans_mask & (Transaction::OUT_OF_ORDER | Transaction::SUSPENDED_Q);
+  // If the caller isn't armed anymore, execution was handled in the txq loop by an earlier poll.
+  // Otherwise, we determine if we can run the caller transaction now.
+  if (!trans || !trans->IsArmedInShard(sid))
+    return;
```

Comment:
Key change, if we're not armed, some other poll already executed us on this shard. Callers always have a positive use_count so `PollExecution` is now safe to be called unconditionally

Reply:
The question is the following: we _can_ detect that we expired before we run the continuation and the loop, and abort early. This way the code is functionally equal to the previous version. Should we do this?